frontend/src/utilities/useWatchIntegrationComponents.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { cloneDeep } from 'lodash-es';
+import { useAppSelector } from '~/redux/hooks';
+import { OdhApplication } from '~/types';
+import { getIntegrationAppEnablementStatus } from '~/services/integrationAppService';
+import { POLL_INTERVAL } from './const';
+import { isIntegrationApp } from './utils';
+
+export const useWatchIntegrationComponents = (
+  components: OdhApplication[],
+): { checkedComponents: OdhApplication[]; isIntegrationComponentsChecked: boolean } => {
+  const [isIntegrationComponentsChecked, setIsIntegrationComponentsChecked] = React.useState(false);
+  const forceUpdate = useAppSelector((state) => state.forceComponentsUpdate);
+  const initForce = React.useRef<number>(forceUpdate);
+  const integrationComponents = React.useMemo(
+    () => components.filter((component) => isIntegrationApp(component)),
+    [components],
+  );
+  const [newComponents, setNewComponents] = React.useState<OdhApplication[]>([]);
+
+  const updateComponentEnablementStatus = (
+    integrationComponentList: OdhApplication[],
+    componentList: OdhApplication[],
+  ) => {
+    const newComponentsList = cloneDeep(componentList);
+
+    integrationComponentList.forEach((component) => {
+      if (component.spec.internalRoute) {
```

Comment:
The `integrationComponentList` is a list of components that has an internal route which starts with `/api/`

Reply:
Then you need to strongly type this -- you're responding to TypeScript with this if-statement because you didn't update the type to fit the need. Perhaps make a temporary type for this file that makes this required and use the `filter` above to [narrow](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates). 

Writing unnecessary logic is dodging the power and maintainability that TS brings.

Reply:
The if statement here is only for `getIntegrationAppEnablementStatus` since the `component.spec` may not has `interalRoute`

Reply:
Okay, this is low priority -- you're not using TypeScript to the full advantage here -- you're just treating it like JavaScript and double checking your work. It's not the end of the world and it would be several TS lines to clean up this if-statement, and that's not necessary to address while you're trying to merge everything.

Go ahead and ignore this; if you want to know more about the problem, feel free to bring it to the channel and I can explain it more.